onnx2pytorch/convert/operations.py

```diff
@@ -1,181 +1,455 @@
+                        # it must be the initial input
+                        else get_init_parameter(buffer_modules, in_op_id, inputs[0])
+                        for in_op_id in node.input
+                    ]
+
+                in_activations = [
+                    in_act for in_act in in_activations if in_act is not None
+                ]
+
+                # store activations for next layer
+                if isinstance(op, Loop):
+                    outputs = op(buffer_modules, activations, *in_activations)
+                    for out_act_name, output in zip(node.output, outputs):
+                        activations[out_op_id] = output
+                        if out_act_name in scan_outputs_names:
+                            scan_outputs[out_act_name].append(output)
+                elif isinstance(op, partial) and op.func == torch.cat:
+                    output = op(in_activations)
+                    activations[out_op_id] = output
+                    if out_op_id in scan_outputs_names:
+                        scan_outputs[out_op_id].append(output)
+                elif isinstance(op, Identity):
+                    # After batch norm fusion the batch norm parameters
+                    # were all passed to identity instead of first one only
+                    output = op(in_activations[0])
+                    activations[out_op_id] = op(output)
+                    if out_op_id in scan_outputs_names:
+                        scan_outputs[out_op_id].append(output)
+                elif isinstance(op, MULTIOUTPUT_TYPES) or (
+                    isinstance(op, COMPOSITE_TYPES)
+                    and any(isinstance(x, MULTIOUTPUT_TYPES) for x in op.modules())
+                ):
+                    outputs = op(*in_activations)
+                    for out_act_name, output in zip(node.output, outputs):
+                        activations[out_act_name] = output
+                        if out_act_name in scan_outputs_names:
+                            scan_outputs[out_act_name].append(output)
+                else:
+                    output = op(*in_activations)
+                    activations[out_op_id] = output
+                    if out_op_id in scan_outputs_names:
+                        scan_outputs[out_op_id].append(output)
+
+            # Prepare for next iteration
+            cond = activations[cond_out_name]
+            i += 1
+            loop_carried_outputs = [
+                activations[aname] for aname in loop_carried_out_names
+            ]
+            activations.update(zip(loop_carried_in_names, loop_carried_outputs))
+
+        # Set outputs to N loop carried final values, followed by K scan outputs
+        outputs = [activations[lcn] for lcn in loop_carried_out_names]
+        for son in scan_outputs_names:
+            outputs.append(torch.cat([so.unsqueeze(dim=0) for so in scan_outputs[son]]))
+        return outputs
+
+
+COMPOSITE_TYPES = (nn.Sequential,)
```

Comment:
Let's move the COMPOSITE_TYPES and others to `operations/__init__.py`
I would also rename them to:
COMPOSITE_LAYERS
STANDARD_LAYERS
MULTIOUTPUT_LAYERS
What do you think?

Reply:
Done! (I moved them to `constants.py` since these require external imports eg torch.nn)